### PR TITLE
Improve(alicloud_ess_scaling_rule): support TargetTrackingScalingRule and StepScalingRule

### PR DESCRIPTION
**Pull request by TalentNing**
TF_ACC=1 go test ./alicloud -v -run=TestAccAlicloudEss -timeout=300m
=== RUN   TestAccAlicloudEssScalingconfigurationsDataSource
--- PASS: TestAccAlicloudEssScalingconfigurationsDataSource (79.86s)
=== RUN   TestAccAlicloudEssScalinggroupsDataSource
--- PASS: TestAccAlicloudEssScalinggroupsDataSource (62.71s)
=== RUN   TestAccAlicloudEssScalingrulesDataSource
--- PASS: TestAccAlicloudEssScalingrulesDataSource (106.96s)
=== RUN   TestAccAlicloudEssAlarm_basic
--- PASS: TestAccAlicloudEssAlarm_basic (181.80s)
=== RUN   TestAccAlicloudEssAlarm_multi
--- PASS: TestAccAlicloudEssAlarm_multi (68.69s)
=== RUN   TestAccAlicloudEssAttachment_update
--- PASS: TestAccAlicloudEssAttachment_update (124.18s)
=== RUN   TestAccAlicloudEssLifecycleHook_basic
--- PASS: TestAccAlicloudEssLifecycleHook_basic (86.94s)
=== RUN   TestAccAlicloudEssNotification_basic
--- PASS: TestAccAlicloudEssNotification_basic (77.79s)
=== RUN   TestAccAlicloudEssScalingConfigurationUpdate
--- PASS: TestAccAlicloudEssScalingConfigurationUpdate (172.44s)
=== RUN   TestAccAlicloudEssScalingConfigurationMulti
--- PASS: TestAccAlicloudEssScalingConfigurationMulti (57.01s)
=== RUN   TestAccAlicloudEssScalingGroup_basic
--- PASS: TestAccAlicloudEssScalingGroup_basic (93.26s)
=== RUN   TestAccAlicloudEssScalingGroup_costoptimized
--- PASS: TestAccAlicloudEssScalingGroup_costoptimized (66.76s)
=== RUN   TestAccAlicloudEssScalingGroup_vpc
--- PASS: TestAccAlicloudEssScalingGroup_vpc (83.52s)
=== RUN   TestAccAlicloudEssScalingGroup_slb
--- PASS: TestAccAlicloudEssScalingGroup_slb (130.62s)
=== RUN   TestAccAlicloudEssVserverGroups_basic
--- PASS: TestAccAlicloudEssVserverGroups_basic (64.03s)
=== RUN   TestAccAlicloudEssScalingRule_basic
--- PASS: TestAccAlicloudEssScalingRule_basic (75.93s)
=== RUN   TestAccAlicloudEssScalingRule_target_tracking_rule_basic
--- PASS: TestAccAlicloudEssScalingRule_target_tracking_rule_basic (76.49s)
=== RUN   TestAccAlicloudEssScalingRule_step_rule_basic
--- PASS: TestAccAlicloudEssScalingRule_step_rule_basic (42.47s)
=== RUN   TestAccAlicloudEssScalingRuleMulti
--- PASS: TestAccAlicloudEssScalingRuleMulti (38.99s)
=== RUN   TestAccAlicloudEssScheduledTask_basic
--- PASS: TestAccAlicloudEssScheduledTask_basic (86.01s)
=== RUN   TestAccAlicloudEssScheduledTask_multi
--- PASS: TestAccAlicloudEssScheduledTask_multi (42.04s)
PASS
ok  	github.com/terraform-providers/terraform-provider-alicloud/alicloud	1818.579s

### CODE REVIEW

File: alicloud/resource_alicloud_ess_scalingrule.go [hunk 1]
```diff
@@ -169,17 +278,95 @@ func resourceAliyunEssScalingRuleUpdate(d *schema.ResourceData, meta interface{}
...
-
+	scalingRuleType := d.Get("scaling_rule_type").(string)
 	if v, ok := d.GetOk("scaling_rule_name"); ok && v.(string) != "" {
 		request.ScalingRuleName = v.(string)
 	}
+	request.ScalingRuleType = d.Get("scaling_rule_type").(string)
 
-	if v, ok := d.GetOk("cooldown"); ok && v.(int) != 0 {
-		request.Cooldown = requests.NewInteger(v.(int))
+	if string(SimpleScalingRule) == scalingRuleType {
```
Comment (Lexsss) [inline alicloud/resource_alicloud_ess_scalingrule.go]: switch case

File: alicloud/resource_alicloud_ess_scalingrule.go [hunk 2]
```diff
@@ -141,20 +206,64 @@ func resourceAliyunEssScalingRuleUpdate(d *schema.ResourceData, meta interface{}
...
-	}
-
 	if d.HasChange("scaling_rule_name") {
 		request.ScalingRuleName = d.Get("scaling_rule_name").(string)
 	}
 
-	if d.HasChange("cooldown") {
-		request.Cooldown = requests.NewInteger(d.Get("cooldown").(int))
+	scalingRuleType := d.Get("scaling_rule_type")
+	if string(SimpleScalingRule) == scalingRuleType {
```
Comment (Lexsss) [inline alicloud/resource_alicloud_ess_scalingrule.go]: switch case

**Comment by Lexsss** [comment]
rename this pr like improve(ess): supported XXX

File: alicloud/resource_alicloud_ess_scalingrule.go (hunk 1 above)
Comment (TalentNing) [reply alicloud/resource_alicloud_ess_scalingrule.go]: done

File: alicloud/resource_alicloud_ess_scalingrule.go (hunk 2 above)
Comment (TalentNing) [reply alicloud/resource_alicloud_ess_scalingrule.go]: done

**Pull request merged**